ros_industrial_cmake_boilerplate/cmake/cmake_tools.cmake

```diff
@@ -239,9 +275,20 @@ macro(generate_package_config)
     endif()
   endif()
 
+  if (NOT ARG_CONFIG_TEMPLATE)
+      if (ARG_EXPORT)
+          make_default_package_config(HAS_TARGETS DEPENDENCIES ${ARG_DEPENDENCIES})
+      else ()
+          make_default_package_config(DEPENDENCIES ${ARG_DEPENDENCIES})
+      endif()
+      set(_pkg_config_template "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}-config.cmake.in")
+  else ()
+      set(_pkg_config_template "${CMAKE_CURRENT_LIST_DIR}/${ARG_CONFIG_TEMPLATE}")
```

Comment:
It would also be nice if this created a variable if ${PACKAGE_NAME}_CONFIG_DEPENDENCIES that you could use in there custom config.

``` cmake
function (create_package_config_depends)
  set(multiValueArgs DEPENDENCIES)
  ... the code above
  set(${PACKAGE_NAME}_CONFIG_DEPENDENCIES ${ricb_pkgconfig}) # may need parent scope
endfunction()
```

Then here call create_package_config_depends().

Reply:
Sorry, I'm not following at all what the goal is here. Is there a particular example you have in mind?

Reply:
If we add the CFG_EXTRAS then my use case here is not need I could just use the CFG_EXTRAS method. I would ignore this request.